### PR TITLE
Uncaught TypeError: Cannot read property 'performance' of null (close #373)

### LINKED ISSUE #373: Uncaught TypeError: Cannot read property 'performance' of null
Reproduces permanently: http://66.55.147.18:13500/history?timestamp=1453462167132&siteUrl=http://www.constantcontact.com/next?s_tnt=56324:4:0

### PR DESCRIPTION
**Pull request by churkin**
\cc @kirovboris @LavrovArtem

### CODE REVIEW

**Comment by churkin** [comment]
@testcafe-build-bot \retest

**Comment by churkin** [comment]
FPR

File: src/utils/url.js [hunk 1]
```diff
@@ -26,6 +24,33 @@ function validateDestUrl (url) {
...
+            isIframe: false,
+            isForm:   false,
+            isScript: false
+        };
+    }
+
+    return {
+        isIframe: resourceType.indexOf('i') !== -1,
+        isForm:   resourceType.indexOf('f') !== -1,
+        isScript: resourceType.indexOf('s') !== -1
```
Comment (inikulin) [inline src/utils/url.js]: Maybe use re tests instead? It will be much faster, e.g.:
```js
/i/.test(resourceType);
```

Comment (churkin) [inline src/utils/url.js]: @inikulin I did a little test, a regular expression lasts longer:

```javascript
var str = "Hammerhead Health Monitor";

function Time(fn, iter) {
   var start = new Date();
   for (var i=0; i<iter; ++i)
      fn();
   var end = new Date();
   console.log(end-start);
}

function IndexMethod() {
   return str.indexOf('d') != -1;
}

function RegexMethod() {
   return /d/.test(str);
}

Time(IndexMethod, 500000)
Time(RegexMethod, 500000)
```

`indexOf` - 32ms
`regex` - 43ms

Comment (inikulin) [inline src/utils/url.js]: This test lacks statistical significance and uses low-res timers. you jsperf.com for benchmarking: https://jsperf.com/regexp-indexof-perf

Comment (churkin) [inline src/utils/url.js]: Then, ideally, we need to store expressions:
```javascript
var iframeRe = /i/;
var formRe   = /f/;
var scriptRe = /s/;

export function parseResourceType (resourceType) {   
    return {
        isIframe: iframeRe.test(resourceType),
        isForm:   formRe.test(resourceType),
        isScript: scriptRe.test(resourceType)
    };
}
```

Comment (inikulin) [inline src/utils/url.js]: Yep, use constants

**Comment by churkin** [comment]
FPR

**Comment by kirovboris** [comment]
lgtm

**Comment by LavrovArtem** [comment]
lgtm

**Pull request merged**